### PR TITLE
Exams Platunov Yurii IM-33

### PR DESCRIPTION
**Pull request by furaizi**

### CODE REVIEW

File: Tasks/between.js [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
-        return '';
-      } else {
-        str = str.substring(0, i);
-      }
-    }
-  }
-  return str;
+'use strict';
+
+let getValueBetween = (str, prefix, suffix) => {
```
Comment (tshemsedinov) [inline Tasks/between.js:5]: Why let?

File: Tasks/between.js [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
+  let prefixStart = str.indexOf(prefix);
+  let suffixStart = str.indexOf(suffix);
```
Comment (tshemsedinov) [inline Tasks/between.js:7]: Why let?

File: Tasks/between.js [hunk 3]
```diff
@@ -1,21 +1,20 @@
...
+
+  if (prefixStart === -1)
+    return '';
+  if (suffixStart === -1)
+    return '';
+
+  let prefixEnd = prefixStart + prefix.length;
+  let substring = str.substring(prefixEnd, suffixStart);
```
Comment (tshemsedinov) [inline Tasks/between.js:15]: Rewrite using just consts )

File: Tasks/words.js [hunk 4]
```diff
@@ -1,41 +1,13 @@
...
-          flag = true;
-        }
-      }
-    }
-  }
-  return number_of_words_in_s;
+'use strict';
+
+let countWords = (str) => {
+  const spaceDelimiter = /\s+/;
```
Comment (tshemsedinov) [inline Tasks/words.js:6]: Rewrite without regexp

File: Tasks/password.js [hunk 5]
```diff
@@ -1,13 +1,19 @@
...
-    key = key + alphabet[Index];
+'use strict';
+
+let generatePassword = (alphabet, passwordLength) => {
+  let password = '';
+  let randomIndex;
+  let randomChar;
+
+  for (let i = 0; i < passwordLength; i++) {
+    randomIndex = Math.floor(Math.random() * alphabet.length);
```
Comment (tshemsedinov) [inline Tasks/password.js:11]: Extract random, decompose

Comment (furaizi) [reply Tasks/password.js:11]: Corrected

File: Tasks/between.js (hunk 1 above)
Comment (furaizi) [reply Tasks/between.js]: Corrected

File: Tasks/words.js (hunk 4 above)
Comment (furaizi) [reply Tasks/words.js]: Corrected

**Pull request closed without merging**